wremnants/muon_efficiencies_smooth.py

```diff
@@ -106,48 +107,62 @@ def make_muon_efficiency_helpers_smooth(filename = data_dir + "/testMuonSF/allSm
     # in this way the name of the (single) effType axis can be used to activate some internal behaviour as needed (for example trigger won't have variations for not triggering lepton)
```

Comment:
"name used to activate" sounds ominous...

Reply:
It refers to the usage in these lines [0].
The base effStat class is general for all steps, but has two separate functions to activate the variations for things up to trigger or isolation (then we define derived classes to deal with isolation or the rest, so that their () operator is the proper function from the base class). The variables with the indices for the eff_type axis must thus be always defined, with an invalid index when the axis name is not found (in which case the missing indices won't be used in fact, but they need a safe initialisation to be general).

I probably have to stress that we use names for the eff_type axis also to read the corresponding bin for the effSyst [1], where all steps are put together, there is no way to get away with using names at some point. But these names should be universal for us, they are used in any script that deals with efficiencies (including the tnp itself), and if we change them randomly at some point then we are just asking for troubles.

[0] 
https://github.com/WMass/WRemnants/blob/f3d6c45712162b8c68589fc9392bdd7de39e6400/wremnants/include/muon_efficiencies_smooth.h#L353-L358

[1] https://github.com/WMass/WRemnants/blob/f3d6c45712162b8c68589fc9392bdd7de39e6400/wremnants/include/muon_efficiencies_smooth.h#L96-L103